gridcapa-core-cc-app/src/main/java/com/farao_community/farao/gridcapa_core_cc/app/postprocessing/RegularOrDcCgmNetworkResolver.java

```diff
@@ -0,0 +1,79 @@
+import com.powsybl.openrao.data.cracapi.State;
+import com.powsybl.openrao.data.cracioapi.CracImporters;
+import com.powsybl.openrao.data.raoresultapi.RaoResult;
+import org.springframework.stereotype.Service;
+
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.time.Instant;
+
+/**
+ * @author Daniel Thirion {@literal <daniel.thirion at rte-france.com>}
+ */
+@Service
+public class RegularOrDcCgmNetworkResolver {
+    private final FileImporter fileImporter;
+    private final MinioAdapter minioAdapter;
+
+    public RegularOrDcCgmNetworkResolver(final FileImporter fileImporter,
+                                         final MinioAdapter minioAdapter) {
+        this.fileImporter = fileImporter;
+        this.minioAdapter = minioAdapter;
+    }
+
+    public Network resolve(final boolean useDcCGmInput,
+                           final InternalCoreCCRequest coreCCRequest) {
+        final Network network;
+        final HourlyRaoResult hourlyRaoResult = coreCCRequest.getHourlyRaoResult();
+        if (useDcCGmInput) {
+            //If DC CGM was used, apply PRA to regular CGM
+            final CgmsAndXmlHeader cgmsAndXmlHeader = fileImporter.importCgmsZip(coreCCRequest.getCgm());
+            final Path cgmPath = cgmsAndXmlHeader.getNetworkPath(Instant.parse(hourlyRaoResult.getRaoRequestInstant()));
+            network = CoreNetworkImporterWrapper.loadNetwork(cgmPath);
+            final Crac crac = importCracFromHourlyRaoRequest(coreCCRequest, network);
+            final RaoResult raoResult = fileImporter.importRaoResult(hourlyRaoResult.getRaoResultFileUrl(), crac);
+            applyRemedialActionsForState(network, raoResult, crac.getPreventiveState());
+        } else {
+            network = fileImporter.importNetworkFromUrl(hourlyRaoResult.getNetworkWithPraUrl());
+        }
+        return network;
+    }
+
+    private static void applyRemedialActionsForState(final Network network,
+                                                     final RaoResult raoResult,
+                                                     final State state) {
+        raoResult.getActivatedNetworkActionsDuringState(state).forEach(networkAction -> networkAction.apply(network));
+        raoResult.getActivatedRangeActionsDuringState(state).forEach(rangeAction ->
+                rangeAction.apply(network, raoResult.getOptimizedSetPointsOnState(state).get(rangeAction)));
+    }
+
+    private Crac importCracFromHourlyRaoRequest(final InternalCoreCCRequest coreCCRequest,
+                                        final Network network) {
+        final HourlyRaoRequest hourlyRaoRequest = coreCCRequest.getHourlyRaoRequest();
+        final String cracFileUrl = hourlyRaoRequest.getCracFileUrl();
+        try (final InputStream cracFileInputStream = minioAdapter.getFile(cracFileUrl)) {
+            return CracImporters.importCrac(Path.of(cracFileUrl).getFileName().toString(), cracFileInputStream, network);
```

Comment:
To retrieve the filename from the url we can also use FilenameUtils.getName(new URL(url).getPath());